core/tee/tee_svc_storage.c

```diff
@@ -626,6 +627,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 		goto err;
 	}
 
+	/* remove the file if it exists*/
+	if (!filedoesnotexist)
+		tee_file_ops.unlink(file);
```

Comment:
With this we'll be left without a the file if the power is cut or something before the rename has completed. rename(2) is supposed to unlink and rename atomically during the right circumstances.